### ISSUE TITLE
support archives/static libraries

### ISSUE THREAD

**Issue by SSE4** [lief-project/LIEF#233]
I found it useful for my needs to parse also archives/static libraries, as well as corresponding object files (e.g. in COFF or ELF formats). at very least, to obtain list of exported and imported symbols.
does it sound as useful addition for this project?
if yes, I may try to prototype this.

**Comment by romainthomas** [comment]
Yes It's completely in the scope of this project so feel free to make a PR for these features

**Comment by mingwandroid** [comment]
So glad to hear this is in-scope!

**Comment by mingwandroid** [comment]
Hello @romainthomas and @SSE4, do either of you know if anyone has done any work on this yet? I may take a go at it if not, hopefully I can figure it out.

**Comment by SSE4** [comment]
@mingwandroid I have started any work yet, feel free to take it

**Comment by mingwandroid** [comment]
Hi @romainthomas, I have been hacking about with some Python code to handle archives (I hope to convert it to C++ and make a PR at some point). I can give you a URL if you wish, the code is currently in a branch in my own fork of `conda-build`.

It works to a degree on Linux. Since an .ar is just an archive of .elf files (with some indexes), to get the final symbols (and rich information about them) I use `lief.elf.parse(raw=ar_file[elf_start_index:elf_end_index])`.

On Windows, calling `lief.PE.parse(raw=ar_file[coff_start_index:coff_end_index])` does not work since a COFF file isn't the same as a PE file. I get the following error:

`'C:\Users\rdonnelly\conda\conda-build\mingw-w64-libz.dll.a\d000103.o' is not an PE`

Do you have any recommendations for how to proceed at this point? Would the code that deals with COFF be callable here?

Many thanks!

**Comment by mingwandroid** [comment]
Hi @romainthomas, thanks. Hopefully I can get this done!

My (very WIP) python code can be seen [here](https://github.com/mingwandroid/conda-build/blob/debug_work_dirs_and_symlink_srcs/conda_build/os_utils/liefldd.py#L462-L627). These line numbers may change today though as I'll be trying to fix/add macOS/BSD support. The function in question is `get_static_lib_exports`. You can see for the non-COFF case that I just let LIEF do the heavy lifting and that's working well.

Thank you for your recommendation. I was under the impression that a COFF file is essentially embedded in a PE file, given what you said, I guess that's not correct?

**Comment by romainthomas** [comment]
Hi @mingwandroid,
Sorry I mess your latest message.  

> Thank you for your recommendation. I was under the impression that a COFF file is essentially embedded in a PE
>  file, given what you said, I guess that's not correct?
PE files implement COFF spec so that a PE file should be COFF file (If I'm not wrong) 

> Are there any channels (gitter/IRC) where hacking on LIEF gets discussed?

Yes there is a Gitter Channel here: https://gitter.im/lief-project

**Comment by SSE4** [comment]
PE and COFF share the same specification, that's true, but COFF is not embedded in PE or vice versa. they have different set of headers, and headers have different set of fields depending if it's COFF or PE file, also some fields may have different meaning.

**Comment by aeflores** [comment]
Is this still being worked on?

**Comment by romainthomas** [comment]
@aeflores not from my side
